src/libpsc/cuda/mparticles_patch_cuda.hxx

```diff
@@ -26,7 +26,16 @@ struct ConstAccessorCuda
     : mprts_{mprts}, data_{const_cast<Mparticles&>(mprts).get_particles()}, off_{mprts.get_offsets()}
   {}
 
-  Patch operator[](int p) const { return {*this, p}; }
+  Patch operator[](int p) const
+  {
+    if(mprts_.need_reorder()) {
+      printf("Warning: Calling accessor with unordered particles! Expect invalid results\n");
+      abort();
+    }
+
+    return {*this, p};
+  }
+
```

Comment:
Hmmh, so weirdly I had lost this piece when I first pushed the PR, but well, I forced it back.

Anyway, the question i have here is that you're essentially adding this checking code in two places, and I'm confused as to why?

Reply:
I only see have it where you listed above, I think the other can go

Reply:
Well, actually as-is this PR is bad, because the thing I added back in made it not compile. Which unfortunately the very limited CI here doesn't catch, since it doesn't compile the cuda part.

Reply:
Oh ok, I was confused when I made the changes to mparticles_cuda.hxx, that change doesn't do anything, I deleted it later

Reply:
If you click on the 'files changed' tab, you can see all the changes.

Reply:
Okay, I guess that makes sense then, as to why it was first there and then disappeared for me in my rebasing endeavor ;)

Reply:
Okay, got rid of it again and force-pushed, so everything should be good.